.travis.yml

```diff
@@ -18,25 +18,30 @@ jobs:
       script:
         - travis/travis.sh
 
-    - name: "Python 2.7 tests"
+    - name: "Python 2.7 + scala 2.11 + spark 2.4 tests"
       language: python
       python: 2.7
       install:
         - pip install "virtualenv<20.0.0" tox  # Bug https://github.com/pypa/virtualenv/issues/1551
       script:
-        - sbt +compile
-        - tox -c python/tox.ini -e py27 -v
+        - sbt -DsparkVersion=2.4.5 -DscalaVersion=2.11.12 compile
+        - tox -c python/tox-spark2.4.ini -e py27 -v
 
-    - name: "Python 3.6 tests"
-      language: python
-      python: 3.6
+    - name: "Python 3.7 + scala 2.11 + spark 2.4 tests"
+      language: python  # This can't be java and python at the same time
+      python: 3.7
+      env:
+        # Required when not setting 'language: java' to force the correct jdk8
+        - PATH=$(echo "$PATH" | sed -e 's/:\/usr\/local\/lib\/jvm\/openjdk11\/bin//')
+        - JAVA_HOME="/usr/lib/jvm/java-1.8.0-openjdk-amd64"
+      dist: xenial  # Travis won't install py37 on trusty
       install:
         - pip install "virtualenv<20.0.0" tox  # Bug https://github.com/pypa/virtualenv/issues/1551
       script:
-        - sbt +compile
-        - tox -c python/tox.ini -e py36 -v
+        - sbt -DsparkVersion=2.4.5 -DscalaVersion=2.11.12 compile
+        - tox -c python/tox-spark2.4.ini -e py37 -v
 
-    - name: "Python 3.7 tests"
+    - name: "Python 3.7 + scala 2.12 + spark 3.0 tests"
```

Comment:
Rather than having this complex matrix running tests to support all of these combinations, I wonder if it makes sense to just upgrade as a clean cut. we could just create an additional point release  moving to spark3.0  (say mleap 0.20) and whoever wants spark2.4 just uses mleap 0.16

but I'd refer to @hollinwilkins  and @ancasarb for more insights

Reply:
We hope mleap is backwards compatible, e.g., user train and saved model by old version mleap, we can load old version model via new version mleap and run model prediction.

Reply:
@lucagiovagnoli If we make two branch mleap, one for spark 2.4.x and another for spark 3.x, then mleap new features and bugfix added in future should go into the two branches code, it make things more troublesome. So that I think best approach will still be make mleap master code support both spark 2.4.x and spark 3.x, in this PR I make a thin wrapper to address spark ML API incompatibility between spark 2.4.x and spark 3.x